src/main/kotlin/apply/application/AdministratorService.kt

```diff
@@ -1,14 +1,38 @@
 package apply.application
 
+import apply.domain.administrator.Administrator
 import apply.domain.administrator.AdministratorRepository
+import apply.domain.administrator.getById
+import apply.domain.administrator.getByUsername
 import org.springframework.security.core.userdetails.UserDetails
 import org.springframework.security.core.userdetails.UserDetailsService
-import org.springframework.security.core.userdetails.UsernameNotFoundException
 import org.springframework.stereotype.Service
 
 @Service
 class AdministratorService(private val administratorRepository: AdministratorRepository) : UserDetailsService {
-    override fun loadUserByUsername(name: String): UserDetails {
-        return administratorRepository.findByName(name) ?: throw UsernameNotFoundException("사용자를 찾을 수 없습니다.")
+    override fun loadUserByUsername(username: String): UserDetails {
+        return administratorRepository.getByUsername(username)
+    }
+
+    fun save(request: AdministratorData): AdministratorResponse {
+        check(!administratorRepository.existsByName(request.name)) { "이미 등록된 관리자명입니다." }
+        check(!administratorRepository.existsByUsername(request.username)) { "이미 등록된 관리자ID 입니다." }
```

Comment:
r: 관리자 이름은 중복이어도 괜찮을 것 같아요.

```suggestion
        check(!administratorRepository.existsByUsername(request.username)) { "이미 등록된 사용자명입니다." }
```